go/types/opcache.go

```diff
@@ -122,6 +180,45 @@ func (i *opCacheIterator) Op() sequenceItem {
 	return entry
 }
 
+// Insert can be called from any goroutine
+func (p *opCache) Insert(val Value) {
+	switch val.Type().Kind() {
+	default:
+		// This is the complicated case. For non-primitives, we want the ldb key to be the hash of mapKey, but we obviously need to get both mapKey and mapVal into ldb somehow. The simplest thing is just to do this:
+		//
+		//     uint32 (4 bytes)             bytes
+		// +-----------------------+----------------------+
+		// | key serialization len |   serialized value   |
```

Comment:
I don't think this is what you're doing, here? You're just setting the LDB key to be `val.Hash()`, essentially, and then encoding val to a byte slice and setting that to be the LDB value